yaml/shared/src/main/scala/org/virtuslab/yaml/Node.scala

```diff
@@ -9,16 +9,34 @@ sealed trait Node:
   def pos: Option[Position]
 
 object Node:
-  final case class ScalarNode(value: String, pos: Option[Position] = None) extends Node
+  final case class ScalarNode private[yaml] (value: String, pos: Option[Position] = None)
+      extends Node
 
-  final case class SequenceNode(nodes: Seq[Node], pos: Option[Position] = None) extends Node
-  object SequenceNode:
-    def apply(nodes: Node*): SequenceNode = SequenceNode(nodes, None)
+  object ScalarNode:
+    given c1: (String => ScalarNode)     = ScalarNode(_)
```

Comment:
This name is ambiguous, you don't have to define the given name here, it will be easier

Reply:
Yeah, it can be anonymous given, couldn't agree more!